src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -138,16 +140,21 @@ public Task StartAsync(ITransport transport, HttpClient httpClient)
 
             _transportChannel = new ChannelConnection<Message>(applicationToTransport, transportToApplication);
 
-            var ignore = Input.Completion.ContinueWith(t =>
+            var ignore = Input.Completion.ContinueWith(async t =>
```

Comment:
Because the channel can be closed from the other side and this is the only way to know it.